File: software/examples/c/example_simple_voltage.c
#include <stdio.h>

#include "ip_connection.h"
#include "bricklet_industrial_analog_out.h"

#define HOST "localhost"
#define PORT 4223
#define UID "XYZ" // Change to your UID

int main() {
	// Create IP connection
	IPConnection ipcon;
	ipcon_create(&ipcon);

	// Create device object
	IndustrialAnalogOut ao;
	industrial_analog_out_create(&ao, UID, &ipcon);

	// Connect to brickd
	if(ipcon_connect(&ipcon, HOST, PORT) < 0) {
		fprintf(stderr, "Could not connect\n");
		exit(1);
	}
	// Don't use device before ipcon is connected

	// Set a voltage of 3.3V
	industrial_analog_out_set_voltage(&ao, 3300);
	industrial_analog_out_enable(&ao);

	printf("Press key to exit\n");
	getchar();
	ipcon_destroy(&ipcon); // Calls ipcon_disconnect internally
}

File: software/examples/go/example_simple_voltage.go
package main

import (
	"fmt"
	"github.com/tinkerforge/go-api-bindings/industrial_analog_out_bricklet"
	"github.com/tinkerforge/go-api-bindings/ipconnection"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Industrial Analog Out Bricklet.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	iao, _ := industrial_analog_out_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Set output voltage to 3.3V
	iao.SetVoltage(3300)
	iao.Enable()

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

	iao.Disable()
}
